sklearn/preprocessing/_function_transformer.py

```diff
@@ -64,11 +64,31 @@ def fit(self, X, y=None):
             check_array(X, self.accept_sparse)
         return self
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length len(input_features), optional
+            String names for input features if available. By default,
+            None is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length len(input_features)
+
+        """
+        if input_features is not None:
+            input_features = list(map(lambda input_feature: 'f(' +
```

Comment:
We've got to balance a few things here:
* a machine-readable description of a feature's provenance (ideally a structured representation)
* a human readable (in limited memory!) description of a feature as distinct from a neighbour
* backwards compatibility issues (although I am willing to label some of these things as experimental and subject to change)